zookeeper-server/src/main/java/org/apache/zookeeper/server/ServerCnxnFactory.java

```diff
@@ -272,4 +277,48 @@ protected void configureSaslLogin() throws IOException {
         }
     }
 
+    /** Maximum number of connections allowed in the ZooKeeper system */
+    public int getMaxCnxns() {
+        return maxCnxns;
+    }
+
+    protected void initMaxCnxns() {
+        String mxCxnProp = System.getProperty(ZOOKEEPER_MAX_CONNECTION);
+        maxCnxns = ZOOKEEPER_MAX_CONNECTION_DEFAULT;
+        if (null != mxCxnProp) {
+            LOG.info("maxCnxns configured value is {}.", mxCxnProp);
+            try {
+                maxCnxns = Integer.parseInt(mxCxnProp.trim());
+            } catch (NumberFormatException e) {
+                LOG.warn("maxCnxns configured value '{}' can not be parsed to integer, using default value {}.",
+                        mxCxnProp, ZOOKEEPER_MAX_CONNECTION_DEFAULT);
+                }
```

Comment:
This can be simplified with `maxCnxns  = Integer.getInteger(ZOOKEEPER_MAX_CONNECTION, ZOOKEEPER_MAX_CONNECTION_DEFAULT)`.

Reply:
Done, thanks